docs/decisions/diagrams/tool-call-auto-invoke.mmd

```diff
@@ -0,0 +1,26 @@
+---
+title: Tool Call with Auto Invoke Kernel Functions
+---
+sequenceDiagram
+    participant Client
+    participant Plugin
+    participant Kernel
+    participant AI Service
+    participant LLM
+    Client->>+AI Service: Invoke Chat Completion with Auto Function Call
+    AI Service->>+LLM: Chat Completion
+    loop For Each Tool LLM Requires
```

Comment:
Does SK perform the "For Each Tool LLM Requires" loop?

Reply:
No this is a loop the LLM may initiate.